component/org.wso2.carbon.identity.api.otp.service.emailotp/src/main/java/org/wso2/carbon/identity/api/otp/service/emailotp/impl/EmailotpApiServiceImpl.java

```diff
@@ -107,4 +110,42 @@ public Response emailotpValidatePost(OTPValidationRequest otpValidationRequest)
             return EndpointUtils.handleUnexpectedServerError(e, log);
         }
     }
+
+    /**
+     * This method is implemented from org.wso2.carbon.identity.api.otp.service.emailotp.EmailotpApi
+     * to verify OTP without invalidate.
+     *
+     * @param otpVerificationRequest Otp verification request object.
+     * @return Response
+     */
+    @Override
+    public Response emailotpVerifyPost(OTPVerificationRequest otpVerificationRequest) {
+
+        String transactionId = StringUtils.trim(otpVerificationRequest.getTransactionId());
+        String userId = StringUtils.trim(otpVerificationRequest.getUserId());
+        String emailOtp = StringUtils.trim(otpVerificationRequest.getEmailOtp());
+        try {
+            ValidationResponseDTO responseDTO = EndpointUtils.getEmailOTPService().verifyEmailOTP(
+                    transactionId, userId, emailOtp);
+            FailureReasonDTO failureReasonDTO = responseDTO.getFailureReason();
+            OTPVerificationFailureReason failureReason = null;
+            if (failureReasonDTO != null) {
```

Comment:
check for `isValid` 1st and then get the failure reason

Reply:
can there be a failure reason even when `isValid` is true?

Reply:
yes it can't be. I used same logic in "/validate" API. We can change the logic.
but this API is not sending empty value in response object 
@JsonInclude(JsonInclude.Include.NON_EMPTY)

Reply:
but at code level we need to have it properly

Reply:
A separate git issue [1] was created as this mentioned issue should be addressed elsewhere as well
[1] https://github.com/wso2/product-is/issues/14491